### PR TITLE
[FSSDK-8948] fix(odp): check odp identifiers not null or empty before sending

### PR DESCRIPTION
**Pull request by mnoman09**
## Summary
- Check if ODP events have at least one key-value pair for identifiers.
If not, discard the request without sending to the ODP server.

## Test plan
- Add a test with empty identifiers and null identifier.
- All tests should pass. 

## Issues
- [FSSDK-8948](https://jira.sso.episerver.net/browse/FSSDK-8948)

### CODE REVIEW

File: OptimizelySDK/Odp/Constants.cs [hunk 1]
```diff
@@ -65,7 +65,12 @@ public static class Constants
         /// <summary>
         /// Default message to log when an ODP Event contains invalid data
         /// </summary>
-        public const string ODP_INVALID_DATA_MESSAGE = "ODP data is not valid.";
+        public const string ODP_INVALID_DATA_MESSAGE = "ODP event send failed (event data is not valid).";
```
Comment (mikechu-optimizely) [inline OptimizelySDK/Odp/Constants.cs:68]: The TDD has a defined message 
odpInvalidData ERROR "ODP data is not valid."

I think we should adhere unless @jaeopt has other input.

File: OptimizelySDK/Odp/Constants.cs [hunk 2]
```diff
@@ -65,7 +65,12 @@ public static class Constants
...
+
+        /// <summary>
+        /// Default message to log when an ODP Identifiers is invalid
+        /// </summary>
+        public const string ODP_INVALID_IDENTIFIERS_MESSAGE = "ODP event send failed (event identifiers must have at least one key-value pair).";
```
Comment (mikechu-optimizely) [inline OptimizelySDK/Odp/Constants.cs:73]: I'm open to adding new (more descriptive) messages as long as @jaeopt 's cool with it.

File: OptimizelySDK/Odp/OdpEventManager.cs [hunk 3]
```diff
@@ -380,6 +386,20 @@ public void UpdateSettings(OdpConfig odpConfig)
...
+        /// </summary>
+        /// <param name="Identifiers">Identifiers to be analyzed</param>
+        /// <returns>True if identifiers dictionary is not null or empty otherwise False</returns>
+        private bool IsIdentifiersValid(Dictionary<string, string> Identifiers) 
+        {
+            if (Identifiers == null || Identifiers.Count == 0)
+            {
+                return false;
+            }
+            return true;
```
Comment (mikechu-optimizely) [inline OptimizelySDK/Odp/OdpEventManager.cs:400]: Would this work?

```csharp
return Identifiers?.Count > 0;
```

**Review by mikechu-optimizely — CHANGES_REQUESTED** [review]
I have some thoughts below.

File: OptimizelySDK/Odp/Constants.cs (hunk 1 above)
Comment (mnoman09) [reply OptimizelySDK/Odp/Constants.cs:68]: event data is separate and identifier is is not part of event data.

**Comment by mikechu-optimizely** [comment]
@mnoman09, This is ready to merge whenever you'd like.
